### PR TITLE
106 adding product and user pagination

### PR DESCRIPTION
**Pull request by muhammad-ahmad45**

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+class ProductsController < ApplicationController
+  before_action :get_params, only: [:edit, :update, :destroy]
```
Comment (DaniyalXprolabs) [inline app/controllers/products_controller.rb:3]: Add an empty line after before action

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    if @product.save
+      redirect_to '/products'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+
```
Comment (DaniyalXprolabs) [inline app/controllers/products_controller.rb]: please write it as
def edit; end

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        <li class="nav-item">
+          <a class="nav-link active" aria-current="page" href="/">Home</a>
+        </li>
+        <li class="nav-item">
+          <a class="nav-link" href="products#index">Products</a>
+        </li>
+      </ul>
+    </div>
+  </div>
+</nav>
```
Comment (DaniyalXprolabs) [inline app/views/layouts/_header.html.erb]: Please add empty line at end of this file

File: test/controllers/products_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require "test_helper"
```
Comment (DaniyalXprolabs) [inline test/controllers/products_controller_test.rb]: Please remove this file

File: .byebug_history [hunk 5]
```diff
@@ -1,3 +1,15 @@
+c
+@users.first.roles
+@users.class
+@users
+q
+user
```
Comment (harissmansoor) [inline .byebug_history]: remove

Comment (muhammad-ahmad45) [reply .byebug_history]: Resolved

**Pull request merged**